components/suggestion/command_provider/app_command_parser/app_command_parser.ts

```diff
@@ -526,6 +548,230 @@ export class ParsedCommand {
+                    break;
+                }
+                default: {
+                    this.incomplete += c;
+                    this.i++;
+                    break;
+                }
+                }
+                break;
+            }
+
+            case ParseState.MultiselectEndTickedValue:
+            case ParseState.MultiselectEndQuotedValue:
+            case ParseState.MultiselectEndValue: {
+                if (!this.field) {
+                    return this.asError(this.intl.formatMessage({
+                        id: 'apps.error.parser.missing_field_value',
+                        defaultMessage: 'Field value is missing.',
+                    }));
+                }
+
+                if (autocompleteMode && c === '') {
+                    return this;
+                }
+                (this.values![this.field.name] as string[]).push(this.incomplete);
+                this.incomplete = '';
+                this.incompleteStart = this.i;
+                if (c === '') {
+                    return this;
+                }
+                this.state = ParseState.MultiselectValueSeparator;
+                break;
+            }
+
+            case ParseState.MultiselectValueSeparator:
+                switch (c) {
+                case '':
+                    if (!autocompleteMode) {
+                        return this.asError(this.intl.formatMessage({
+                            id: 'apps.error.parser.missing_list_end',
+                            defaultMessage: 'Expected list closing token.',
+                        }));
+                    }
+                    return this;
+                case ']':
+                    this.i++;
+                    this.state = ParseState.ParameterSeparator;
+                    break;
+                case ' ':
+                case '\t':
+                    this.i++;
+                    break;
+                case ',':
+                    this.i++;
+                    this.state = ParseState.MultiselectNextValue;
+                    break;
+                default:
+                    return this.asError(this.intl.formatMessage({
+                        id: 'apps.error.parser.unexpected_character',
+                        defaultMessage: 'Unexpected character.',
+                    }));
+                }
+                break;
+            case ParseState.MultiselectNextValue:
+                switch (c) {
+                case ' ':
+                case '\t':
+                    this.i++;
+                    break;
+                default:
+                    this.state = ParseState.MultiselectStartValue;
+                }
+                break;
+            default:
+                return this.asError(this.intl.formatMessage({
+                    id: 'apps.error.parser.unexpected_state',
+                    defaultMessage: 'Unreachable: Unexpected state in matchBinding: `{state}`.',
```

Comment:
I think we're trying to limit the number of low-level error messages going forward. Can we instead log the error and display a more generic error?

Reply:
Being an unreachable error message (this should never happen to a end user) I think it is ok having it as a low-level message. @aaronrothschild thoughts?

Reply:
I'm not sure if the likelihood of the error changes the principle. We are sort of polluting the `i18n.json` file with errors that are not very helpful. @enahum may have some input on this

Reply:
Should we make this error not translated?

Reply:
If we haven't added it to the `en.json` in a previous PR, I think we should avoid adding it here. We never arrived at a full plan to handle this, discussed briefly here https://community-daily.mattermost.com/core/pl/jg79zt5otpnuzccdo79wpb75gy

Reply:
These are already in place and translated. We can remove them, but I think it would be better in a different PR.
The strings added in this PR are all user facing.